src-common/vcastServer.ts

```diff
@@ -1,6 +1,9 @@
 // proof of concept for interacting with a running environment data server
 
-import fetch from "node-fetch";
+const fetch = (...args: [string, RequestInit?]) =>
```

Comment:
What is this and why is it changing in a PR to fix compile errors in UTs

Reply:
```
node_modules/vite/node_modules/esbuild/lib/main.js:1373:27: ERROR: [plugin: externalize-deps] "node-fetch" resolved to an ESM file. ESM file cannot be loaded by `require`. See http://vitejs.dev/guide/troubleshooting.html#this-package-is-esm-only for more details.
```

https://stackoverflow.com/a/69331469

Reply:
https://github.com/node-fetch/node-fetch/blob/HEAD/docs/v3-UPGRADE-GUIDE.md

```
const fetch = (...args) => import('node-fetch').then(({default: fetch}) => fetch(...args));
```

Reply:
To be clear: I don't know the difference between how the tests run and how the extension is built, and why this seems okay when building the extension.

Reply:
Okay, got it. The problem is that we're trying to share `getToolVersion` into `vite.config.ts`, which causes this issue.

@Den1552: let's put `getToolVersion` in its own file and minimise what gets imported into that new file. Hopefully that solves this problem.

Reply:
Refactored in 8fdcb1cdf2798b7189e78e16f572f035c8fa8f64 .